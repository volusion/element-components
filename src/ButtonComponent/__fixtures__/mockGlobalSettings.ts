import { GlobalSettings } from '../types';
export const mockGlobalSettings: GlobalSettings = {
    globalComponents: {
        volComponentButton: {
            primaryButtonStyles: {
                size: 'large',
                rounded: 'none',
                textColor: '#fff',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: true,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: '#fff',
                backgroundColor: 'rgba(247, 68, 186, 1)',
                borderThickness: 'none',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(218, 15, 148, 1)'
            },
            secondaryButtonStyles: {
                size: 'block',
                rounded: 'none',
                textColor: 'rgba(255, 255, 255, 1)',
                fontWeight: '700',
                borderColor: 'rgba(247, 67, 186, 1)',
                growOnHover: false,
                letterSpacing: 'none',
                textTransform: 'none',
                hoverTextColor: 'rgba(255, 255, 255, 1)',
                backgroundColor: 'rgba(247, 74, 201, 1)',
                borderThickness: 'thin',
                hoverBorderColor: 'rgba(0, 0, 0, 1)',
                hoverBackgroundColor: 'rgba(0, 0, 0, 1)'
            }
        }
    }
};
