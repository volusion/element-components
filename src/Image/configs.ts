import { ElementPropTypes } from '@volusion/element-proptypes';

export const defaultConfig = {
    alt: '',
    height: 270,
    src:
        'https://res.cloudinary.com/dyx4yhvoq/image/upload/v1545428185/images/tcp-no-image.jpg',
    title: '',
    width: 360
};

export const configSchema = {
    src: {
        label: 'Image',
        type: ElementPropTypes.string
    },
    title: {
        label: 'Title',
        type: ElementPropTypes.string
    },
    altText: {
        label: 'Alt Text',
        type: ElementPropTypes.string
    },
    width: {
        label: 'Width',
        type: ElementPropTypes.number
    },
    height: {
        label: 'Height',
        type: ElementPropTypes.number
    }
};
