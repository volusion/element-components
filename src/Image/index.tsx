import React, { CSSProperties } from 'react';
import { BlockUtils, GlobalSettings } from '../types';

interface ImgProps {
    alt: string;
    height: number;
    src: string;
    title: string;
    width: number;
}

interface AmpImgProps extends ImgProps {
    layout: string;
}

export interface ImageProps extends ImgProps {
    globalSettings: GlobalSettings;
    utils: BlockUtils;
}

const responsiveImgStyles: CSSProperties = {
    maxWidth: '100%'
};

const AmpImg = (props: AmpImgProps) => <amp-img {...props} />;
const Img = (props: ImgProps) => (
    <picture>
        <img {...props} style={responsiveImgStyles} />
    </picture>
);

const Image = ({ alt, height, src, title, utils, width }: ImageProps) => {
    const imgProps = { alt, height, src, title, width };

    return utils.isAmpRequest ? (
        <AmpImg layout="responsive" {...imgProps} />
    ) : (
        <Img {...imgProps} />
    );
};

export const block = Image;
export { configSchema, defaultConfig } from './configs';
