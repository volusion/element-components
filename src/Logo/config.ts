import { ElementPropTypes } from '@volusion/element-proptypes';
import { LogoStyle } from './index';

export const defaultConfig = {
  style: 'Text' as LogoStyle,
};

export const configSchema = {
  style: {
    label: 'Style',
    type: ElementPropTypes.oneOf(['Text', 'Image']),
  },
  Image: {
    label: 'Image',
    type: ElementPropTypes.component('Image'),
  },
  StoreName: {
    label: 'Store Name',
    type: ElementPropTypes.component('Text'),
  }
};