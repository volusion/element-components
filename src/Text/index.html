<html>
  <head>
    <link
      rel="stylesheet"
      href="https://unpkg.com/tachyons@4.10.0/css/tachyons.min.css"
    />
    <script
      crossorigin
      src="https://unpkg.com/react@16/umd/react.development.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/aphrodite@2.2.1/dist/aphrodite.umd.min.js"
    ></script>
    <script src="https://sdk.v2-prod.volusion.com/element-sdk.umd.js"></script>
    <script src="../../dist/component.umd.js"></script>
  </head>
  <body>
    <div id="root"></div>
  </body>
  <script>
    var globalSettings = {
      color: {
        background: "#fff",
        link: "rgba(51, 51, 51, 1)",
        linkHover: "rgba(235, 16, 148, 1)",
        primary: "#e53d2d",
        salePrice: "rgba(235, 16, 148, 1)",
        secondary: "#333",
        text: "#000"
      },
      globalComponents: {
        volComponentButton: {
          primaryButtonStyles: {
            size: "large",
            rounded: "none",
            textColor: "#fff",
            fontWeight: "700",
            borderColor: "rgba(247, 67, 186, 1)",
            growOnHover: true,
            letterSpacing: "none",
            textTransform: "none",
            hoverTextColor: "#fff",
            backgroundColor: "rgba(247, 68, 186, 1)",
            borderThickness: "none",
            hoverBorderColor: "rgba(0, 0, 0, 1)",
            hoverBackgroundColor: "rgba(218, 15, 148, 1)"
          },
          secondaryButtonStyles: {
            size: "block",
            rounded: "none",
            textColor: "rgba(255, 255, 255, 1)",
            fontWeight: "700",
            borderColor: "rgba(247, 67, 186, 1)",
            growOnHover: false,
            letterSpacing: "none",
            textTransform: "none",
            hoverTextColor: "rgba(255, 255, 255, 1)",
            backgroundColor: "rgba(247, 74, 201, 1)",
            borderThickness: "thin",
            hoverBorderColor: "rgba(0, 0, 0, 1)",
            hoverBackgroundColor: "rgba(0, 0, 0, 1)"
          }
        }
      },
      typography: {
        baseFontSize: "16px",
        fontFamily: '"Lato", sans-serif',
        headingFontFamily: '"Oswald", sans-serif',
        headingWeight: "normal",
        lineHeight: "1.15"
      }
    };

    var ElementPropTypes = window.ElementSdk.ElementPropTypes;

    var element = React.createElement(
      ElementComponents.Text.block,
      {
          content: '<h1>Text Component</h1>',
          globalSettings 
      }
    );

    ReactDOM.render(
      element,
      document.getElementById("root")
    );
  </script>
</html>
