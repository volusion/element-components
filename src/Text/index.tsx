import React from 'react';
import { ElementPropTypes } from '@volusion/element-proptypes';
import { ComponentProps } from '../../types';

interface TextProps extends ComponentProps {
    htmlString: string;
}

const Text: React.FC<TextProps> = ({ htmlString, ...props }) => {
    if (!htmlString) return null;
    // We delete the unused globalSettings prop so that React won't complain about it not being a valid DOM attribute.
    delete props.globalSettings;
    return <div {...props} dangerouslySetInnerHTML={{ __html: htmlString }} />;
};

export const block = Text;

export const defaultConfig = {
    htmlString: 'Lorem Ipsum'
};

export const configSchema = {
    htmlString: {
        label: 'Text Content',
        type: ElementPropTypes.editorMinimal
    }
};
