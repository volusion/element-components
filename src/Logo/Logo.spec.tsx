import React from 'react';
import { mount } from 'enzyme';
import { block as Logo, LogoProps } from './index';
import { defaultConfig } from './config';
import { block as Text } from '../Text';
import { block as Image } from '../Image';
import { mockGlobalSettings as globalSettings } from '../__fixtures__/mockGlobalSettings';

describe('Logo component', () => {
    let props: LogoProps;

    beforeEach(() => {
        props = {
            ...defaultConfig,
        };
    });

    it('renders a text logo', () => {
        const textProps = {
            globalSettings,
            htmlString: 'Test Store Name',
        };
        const TextComponent = () => <Text {...textProps}  />;
        props.StoreName = TextComponent;

        const renderedTextLogo = mount(<Logo {...props} />);
        expect(renderedTextLogo.render()).toMatchSnapshot();
    });

    it('renders an image logo', () => {
        const imageProps = {
            globalSettings,
            src: 'http://example.com/test.jpg',
            alt: '',
            title: '',
            height: 10,
            width: 10,
            utils: {
                isAmpRequest: false
            }
        };
        const ImageComponent = () => <Image {...imageProps} />;
        props.style = 'Image';
        props.Image = ImageComponent;

        const renderedImageLogo = mount(<Logo {...props} />);
        expect(renderedImageLogo.render()).toMatchSnapshot();
    });
});