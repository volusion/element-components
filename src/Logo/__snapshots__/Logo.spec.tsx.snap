// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders a text logo 1`] = `
<div>
  Test Store Name
</div>
`;

exports[`Logo component renders an image logo 1`] = `
<picture>
  <img
    alt=""
    height="10"
    src="http://example.com/test.jpg"
    style="max-width: 100%;"
    title=""
    width="10"
  />
</picture>
`;
